pyccel/ast/numpyext.py

```diff
@@ -786,25 +786,44 @@ def __new__(cls, a, dtype=None, order='K', subok=True, shape=None):
 class NumpyNorm(PyccelInternalFunction):
     """ Represents call to numpy.norm"""
     _dtype = NativeReal()
-
-    def __init__(self, arg, dim=None):
-        if isinstance(dim, ValuedArgument):
-            dim = dim.value
-        if self.dim is not None:
-            sh = list(sh)
-            del sh[self.dim]
+    _shape = ()
+    _rank  = 0
+
+    def __init__(self, arg, axis=None):
+        super().__init__(arg, axis)
+        cmplx_precision = None
+        if isinstance(arg.dtype, NativeBool):
+            arg = PythonFloat(arg)
+        elif isinstance(arg.dtype, NativeComplex):
+            cmplx_precision = arg.precision
+            arg = PythonAbs(arg)
```

Comment:
I see that you store the precision as `PythonAbs` is a builtin function which therefore always returns an object with the default precision. Could you not get round this problem by using `NumpyAbs`?

Reply:
`NumpyAbs` return the same as `PythonAbs`.

Reply:
Ah sorry, you are right. But they also both preserve the precision:
```
In [2]: type(abs(np.float32(3)))
Out[2]: numpy.float32

In [3]: type(np.abs(np.float32(3)))
Out[3]: numpy.float32

In [4]: type(np.abs(np.complex64(3)))
Out[4]: numpy.float32

In [5]: type(abs(np.complex64(3)))
Out[5]: numpy.float32
```